mediapipe/tasks/ios/test/vision/core/MPPImageTests.m

```diff
@@ -0,0 +1,328 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/common/sources/MPPCommon.h"
+#import "mediapipe/tasks/ios/vision/core/sources/MPPImage.h"
+
+#import <Accelerate/Accelerate.h>
+#import <CoreGraphics/CoreGraphics.h>
+#import <CoreMedia/CoreMedia.h>
+#import <CoreVideo/CoreVideo.h>
+#import <XCTest/XCTest.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *const kTestImageName = @"burger";
+static NSString *const kTestImageType = @"jpg";
+static CGFloat kTestImageWidthInPixels = 480.0f;
+static CGFloat kTestImageHeightInPixels = 325.0f;
+static NSString *const kExpectedErrorDomain = @"com.google.mediapipe.tasks";
+
+#define AssertEqualErrors(error, expectedError)                                               \
```

Comment:
FYI (for future PRs): The last internal reviewer asked us to create functions and not rely on macros so much:

"Avoid macros, especially where const variables, enums, Xcode snippets, or C functions may be used instead.

Macros make the code you see different from the code the compiler sees. Modern C renders traditional uses of macros for constants and utility functions unnecessary. Macros should only be used when there is no other solution available."

Macros are only really supposed to be used as compile-time guards.

No need to fix for any current PRs/current code (though a cleanup at a later point would be nice).